torch_xla_py/xla_dist.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""Tool to distribute training on Cloud TPU Pods."""
+from __future__ import division
+from __future__ import print_function
+
+import abc
+
+class BaseWorker(abc.ABC):
+
+  @classmethod
+  @abc.abstractmethod
+  def validate_cluster(cls, workers):
+    """Validates the current worker cluster configuration.
+
+    Raises:
+      RuntimeError: If the cluster is misconfigured, this validation will
+      raise a error.
+    """
+    raise NotImplementedError()
+
+
+class CpuWorker(BaseWorker):
+
+  def __init__(self,
+               internal_ip,
+               machine_type,
+               zone,
+               hostname=None):
+    self._internal_ip = internal_ip
+    self._machine_type = machine_type
+    self._zone = zone
+    self._hostname = hostname
+
+  @classmethod
+  def validate_cluster(cls, workers):
```

Comment:
We should have a `Cluster` entity, which includes Client/Service work sets, which has a `validate()` method.

Reply:
Done.